### PR TITLE
Update build process, fix warnings, improve support for router API, cleanup

### PR DESCRIPTION
**Pull request by devinivy**
- Support single top-level route.
 - Remove lifecycle wrapper component when unused to cleanup the component tree and to address warnings about use of `componentWillUnmount()`.
 - Support react-router's `sensitive` route option.
 - Support react-router's `render` route option.
 - Updated build process to match https://github.com/BigRoomStudios/strange-middle-end.
 - Declare peer dependencies.
 - Depend on react-router rather than react-router-dom to be more generic.
 - Minor cleanup.

In doing this I learned that the builds for strange-middle-end and strange-router have some duplicate helper code generated during the build process.  Would be great to get @babel/runtime working for us.  I gave it a shot but didn't have luck right away.  https://babeljs.io/docs/en/next/babel-plugin-transform-runtime

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -35,116 +34,85 @@ internals.renderRoute = (basePath) => {
...
             }
 
-            if (typeof redirectClone.to === 'string') {
+            if (typeof to === 'string') {
                 // If redirect.to is absolute, leave it be. Otherwise make it relative
-                redirectClone.to = redirectClone.to.startsWith('/') ? redirectClone.to : internals.concatPaths(basePath, redirectClone.to);
+                redirectProps.to = to.startsWith('/') ? to : internals.concatPaths(basePath, to);
             }
-            else {
+            else if (to && typeof to.pathname === 'string') {
```
[the remark this answers is not in the document]
Comment (devinivy) [reply lib/index.js]: We just wouldn't touch/change it!  The way I think of this is that we transform props that we recognize, and we pass the rest through.  If they're invalid, that's not our problem: they're going into the `Redirect` component whose job it is to complain.

File: lib/index.js [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
-        );
+        return Array.isArray(routes) ?
+            <Switch>{routes.map(renderRoute)}</Switch> :
+            renderRoute(route);
     }
 };
 
+exports.Routes.propTypes = {
+    routes: T.oneOfType([
+        T.object,
```
[the remark this answers is not in the document]
Comment (zemccartney) [reply lib/index.js]: ay, yea, that makes sense. thanks! 🙏

File: lib/index.js [hunk 3]
```diff
@@ -1,30 +1,29 @@
...
+        const renderRoute = internals.renderRoute('/');
 
-        return (
-            <Switch>
-                {routes.map(internals.renderRoute('/'))}
-            </Switch>
-        );
+        return Array.isArray(routes) ?
+            <Switch>{routes.map(renderRoute)}</Switch> :
+            renderRoute(route);
```
Comment (bmleight) [inline lib/index.js]: should this be `renderRoute(routes);`?

Comment (devinivy) [reply lib/index.js]: Yeah, I think so!
```suggestion
            renderRoute(routes);
```

**Pull request merged**